ouroboros-consensus/tools/db-convert/Main.hs

```diff
@@ -0,0 +1,275 @@
+  :: EpochSlots
+  -> Path Abs File -- ^ Input
+  -> Path Abs Dir -- ^ Ouput directory
+  -> IO (Either CC.ParseError ())
+convertEpochFile es inFile outDir =
+  let inStream = CC.parseEpochFileWithBoundary es (toFilePath inFile)
+      dbDir = outDir </> [reldir|immutable|]
+      encode = CB.serializeEncoding' . Byron.encodeByronBlock . Byron.ByronBlockOrEBB
+   in do
+        createDirIfMissing True dbDir
+        outFileName <-
+          parseRelFile $ "epoch-" <>
+            drop 2 (toFilePath (filename inFile))
+        outFile <- (dbDir </> outFileName) -<.> "dat"
+        IO.withFile (toFilePath outFile) IO.WriteMode $ \h ->
+          runResourceT $ runExceptT $ S.mapM_ (liftIO . BS.hPut h) . S.map encode $ inStream
+
+validateChainDb
+  :: forall blk. (blk ~ ByronBlockOrEBB ByronConfig)
+  => Path Abs Dir -- ^ DB directory
+  -> CC.Genesis.Config
+  -> Bool -- Verbose
+  -> IO ()
+validateChainDb dbDir cfg verbose =
+  do
+    initialCVS <- runExceptT $ CC.initialChainValidationState cfg
+    case initialCVS of
+      Left err -> throwIO $ InitialChainStateError err
+      Right cvs ->
+        withThreadRegistry $ \registry -> do
+          chaindb <- give protocolMagicId $ give epochSlots $ ChainDB.openDB $ args registry
+          blk <- ChainDB.getTipBlock chaindb
+          putStrLn $ "DB tip: " ++ condense blk
+          ChainDB.closeDB chaindb
+        where
+          initLedgerState = ExtLedgerState
+            { ledgerState = Byron.ByronEBBLedgerState $
+                Byron.ByronLedgerState cvs Seq.empty
+            , ouroborosChainState = Seq.empty
+            }
+          protocolMagicId = CB.unAnnotated . getAProtocolMagicId $ CC.Genesis.configProtocolMagic cfg
+          epochSlots = CC.Genesis.configEpochSlots cfg
+          securityParam = SecurityParam $ CC.unBlockCount k
+          k = CC.Genesis.configK cfg
+          args registry =
+            (ChainDB.defaultArgs @blk (toFilePath dbDir))
+              { ChainDB.cdbGenesis = return initLedgerState
+              , ChainDB.cdbDecodeBlock = Byron.decodeByronBlock epochSlots
+              , ChainDB.cdbDecodeChainState = Byron.decodeByronChainState
+              , ChainDB.cdbDecodeHash = Byron.decodeByronHeaderHash
+              , ChainDB.cdbDecodeLedger = Byron.decodeByronLedgerState
+              , ChainDB.cdbEncodeBlock = Byron.encodeByronBlock
+              , ChainDB.cdbEncodeChainState = Byron.encodeByronChainState
+              , ChainDB.cdbEncodeHash = Byron.encodeByronHeaderHash
+              , ChainDB.cdbEncodeLedger = Byron.encodeByronLedgerState
+              , -- Policy
+              ChainDB.cdbValidation = ImmDB.ValidateAllEpochs
+              , ChainDB.cdbBlocksPerFile = 10
+              , ChainDB.cdbMemPolicy = LgrDB.defaultMemPolicy securityParam
+              , ChainDB.cdbDiskPolicy = LgrDB.defaultDiskPolicy securityParam 20000
+              , -- Integration
+              ChainDB.cdbNodeConfig = WithEBBNodeConfig $ EncNodeConfig
```

Comment:
Would it be easier to reuse `protocolInfoByron` here? https://github.com/input-output-hk/ouroboros-network/blob/5a9bd9c8393a079d6f95ba66a50640135f8fdcf2/ouroboros-consensus/src/Ouroboros/Consensus/Node/ProtocolInfo/Byron.hs#L114
Also see
https://github.com/input-output-hk/ouroboros-network/blob/e8849236663a9c1e0f9f81e3915f1b612e0a03df/ouroboros-consensus/src/Ouroboros/Consensus/Protocol.hs#L80